spring-kafka/src/main/java/org/springframework/kafka/core/reactive/ReactiveKafkaConsumerTemplate.java

```diff
@@ -0,0 +1,158 @@
+ *
+ * @param <K> the key type.
+ * @param <V> the value type.
+ *
+ * @author Mark Norkin
+ */
+public class ReactiveKafkaConsumerTemplate<K, V> {
+	private final KafkaReceiver<K, V> kafkaReceiver;
+
+	public ReactiveKafkaConsumerTemplate(ReceiverOptions<K, V> receiverOptions) {
+		this.kafkaReceiver = KafkaReceiver.create(receiverOptions);
+	}
+
+	public Flux<ReceiverRecord<K, V>> receive() {
+		return this.kafkaReceiver.receive();
+	}
+
+	public Flux<ConsumerRecord<K, V>> receiveAutoAck() {
+		return this.kafkaReceiver.receiveAutoAck().flatMap(Function.identity());
+	}
+
+	public Flux<ConsumerRecord<K, V>> receiveAtmostOnce() {
+		return this.kafkaReceiver.receiveAtmostOnce();
+	}
+
+	public Flux<Flux<ConsumerRecord<K, V>>> receiveExactlyOnce(TransactionManager transactionManager) {
+		return this.kafkaReceiver.receiveExactlyOnce(transactionManager);
+	}
+
+	public <T> Mono<T> doOnConsumer(Function<Consumer<K, V>, ? extends T> function) {
+		return this.kafkaReceiver.doOnConsumer(function);
+	}
+
+	public Flux<TopicPartition> assignment() {
+		Mono<Set<TopicPartition>> partitions = doOnConsumer(Consumer::assignment);
+		return partitions.flatMapIterable(Function.identity());
+	}
+
+	public Flux<String> subscription() {
+		Mono<Set<String>> subscriptions = doOnConsumer(Consumer::subscription);
+		return subscriptions.flatMapIterable(Function.identity());
+	}
+
+	public Mono<Void> seek(TopicPartition partition, long offset) {
+		return doOnConsumer(consumer -> {
+			consumer.seek(partition, offset);
+			return null;
+		});
+	}
+
+	public Mono<Void> seekToBeginning(TopicPartition... partitions) {
+		return doOnConsumer(consumer -> {
+			consumer.seekToBeginning(Arrays.asList(partitions));
+			return null;
+		});
+	}
+
+	public Mono<Void> seekToEnd(TopicPartition... partitions) {
+		return doOnConsumer(consumer -> {
+			consumer.seekToEnd(Arrays.asList(partitions));
+			return null;
+		});
+	}
+
+	public Mono<Long> position(TopicPartition partition) {
+		return doOnConsumer(consumer -> consumer.position(partition));
+	}
+
+	public Mono<OffsetAndMetadata> committed(TopicPartition partition) {
+		return doOnConsumer(consumer -> consumer.committed(partition));
+	}
+
+	public Flux<PartitionInfo> partitionsFromConsumerFor(String topic) {
+		Mono<List<PartitionInfo>> partitions = doOnConsumer(c -> c.partitionsFor(topic));
+		return partitions.flatMapIterable(Function.identity());
+	}
+
+	public Flux<TopicPartition> paused() {
+		Mono<Set<TopicPartition>> paused = doOnConsumer(Consumer::paused);
+		return paused.flatMapIterable(Function.identity());
+	}
+
+	public Mono<Void> pause(TopicPartition... partitions) {
+		return doOnConsumer(c -> {
+			c.pause(Arrays.asList(partitions));
+			return null;
+		});
+	}
+
+	public Mono<Void> resume(TopicPartition... partitions) {
+		return doOnConsumer(c -> {
+			c.resume(Arrays.asList(partitions));
+			return null;
+		});
+	}
+
+	public Flux<Map.Entry<MetricName, ? extends Metric>> metricsFromConsumer() {
```

Comment:
I would for the `Tuple` from Reactor is better, than `Map.Entry` to have here.

Reply:
Done. Also changed for other api's on producer side that return Map.Entry.